sklearn/neighbors/binary_tree.pxi

```diff
@@ -1524,6 +1536,11 @@ cdef class BinaryTree:
         cdef DTYPE_t dist_LB = 0, dist_UB = 0
 
         cdef ITYPE_t n_samples = self.data.shape[0]
+        cdef DTYPE_t Z
+        if self.sample_weight is not None:
+            Z = self.sum_weight
```

Comment:
Why do you need both Z and sum_weight?

Reply:
`Z` is either `sum_weight` or `n_samples`, so in this context `sum_weight` is just as useful as `n_samples` although much costlier to produce (hence better to compute it just once at `__init__` time).

Reply:
That would make sense indeed, thanks for the suggestion !